sale_product_set/wizard/product_set_add.py

```diff
@@ -10,37 +10,68 @@ class ProductSetAdd(models.TransientModel):
     _rec_name = 'product_set_id'
     _description = "Wizard model to add product set into a quotation"
 
+    order_id = fields.Many2one(
+        'sale.order', 'Sale Order', required=True,
+        default=lambda self: self.env.context.get('active_id')
+    )
     product_set_id = fields.Many2one(
         'product.set', 'Product set', required=True)
     quantity = fields.Float(
         digits=dp.get_precision('Product Unit of Measure'), required=True,
         default=1)
 
+    def _check_partner(self):
```

Comment:
@simahawk Why not defining a constrains on order_id? With a constrains you don't need to explicitly call this check into the `add_set` method. In the same time, the `onchange` on the product_set_id should not be required is we set the right domain on the product_set_id. (To help we could add a read only related field for `order_id.partner_id`, The domain into the view on product_set_id will be `[ '|', ('partner_id', '=', False), ('partner_id', '=', partner_id)]` )

Reply:
good idea!
I removed the onchange but I left it for the wizard action because you could call it w/out noticing you are passing wrong data.

Reply:
@simahawk No `@constrains("partner_id")`